jetbrains-rider/protocol/model/psi/LambdaPsiModel.kt

```diff
@@ -3,27 +3,95 @@
 
 package protocol.model.psi
 
+import java.io.File
 import com.jetbrains.rd.generator.nova.Ext
 import com.jetbrains.rd.generator.nova.async
-import com.jetbrains.rd.generator.nova.call
+import com.jetbrains.rd.generator.nova.csharp.CSharp50Generator
 import com.jetbrains.rd.generator.nova.doc
 import com.jetbrains.rd.generator.nova.field
+import com.jetbrains.rd.generator.nova.immutableList
+import com.jetbrains.rd.generator.nova.kotlin.Kotlin11Generator
+import com.jetbrains.rd.generator.nova.nullable
+import com.jetbrains.rd.generator.nova.property
+import com.jetbrains.rd.generator.nova.setting
+import com.jetbrains.rd.generator.nova.sink
+import com.jetbrains.rd.generator.nova.source
+import com.jetbrains.rd.generator.nova.util.syspropertyOrInvalid
+import com.jetbrains.rd.generator.nova.ExternalGenerator
+import com.jetbrains.rd.generator.nova.GeneratorBase
+import com.jetbrains.rd.generator.nova.FlowTransform
 import com.jetbrains.rd.generator.nova.PredefinedType.bool
 import com.jetbrains.rd.generator.nova.PredefinedType.int
 import com.jetbrains.rd.generator.nova.PredefinedType.string
+import com.jetbrains.rider.model.nova.ide.IdeRoot
+import com.jetbrains.rider.model.nova.ide.ShellModel.IconModel
 import com.jetbrains.rider.model.nova.ide.SolutionModel
 
+object PsiKotlinGenerator : ExternalGenerator(
+    Kotlin11Generator(FlowTransform.AsIs, "com.jetbrains.rider.model", File(syspropertyOrInvalid("ktGeneratedOutput"))),
+    IdeRoot)
+
+object PsiCSharpGenerator : ExternalGenerator(
+    CSharp50Generator(FlowTransform.Reversed, "JetBrains.Rider.Model", File(syspropertyOrInvalid("csPsiGeneratedOutput"))),
+    IdeRoot)
+
 @Suppress("unused")
 object LambdaPsiModel : Ext(SolutionModel.Solution) {
```

Comment:
should the protocol be a source set so IDE sets up dependencies right?

Reply:
I'm sorry, not sure I got the question. Model in this context is used to generate classes that are used on both side from the protocol: Kotlin – protocol – C#.

Reply:
Well... yes, if I correctly understand your question, you might say it that way. The way you define an exact model - set up how the protocol works inside IDE.

Reply:
How did you set it up in the IDE? Are you editing the protocol in rider or idea? For me the classpath isn't set up so it is missing the kotlin sdk and Rd depenedency

Reply:
I mean from a Gradle perspective